mava/components/building/parameter_client.py

```diff
@@ -116,6 +117,32 @@ def on_building_executor_parameter_client(self, builder: SystemBuilder) -> None:
         params["norm_params"] = builder.store.norm_params
         get_keys.append("norm_params")
 
+        # Create best performance network params in case of evaluator
```

Comment:
I think this method is getting quite long and would benefit from being broken up into smaller methods. This would also help when implementing other algorithms (such as value based ones) as those don't have a policy and critic network, so we could just override the smaller methods and leave most of the functionality untouched. 

^Just a comment for next time, don't make these changes here.

Reply:
We can handle it in this PR, it's fine I will make the changes!

Reply:
I created an issue for this request https://github.com/instadeepai/Mava/issues/831